src/languages/ruby.ts

```diff
@@ -0,0 +1,169 @@
+const EXPRESSION_TYPES = [
+  "array",
+  "assignment",
+  "begin",
+  "binary",
+  "break",
+  "call",
+  "case",
+  "case_match",
+  "chained_string",
+  "character",
+  "class",
+  "class_variable",
+  "complex",
+  "conditional",
+  "constant",
+  "delimited_symbol",
+  "element_reference",
+  "false",
+  "float",
+  "for",
+  "global_variable",
+  "hash",
+  "heredoc_beginning",
+  "identifier",
+  "if",
+  "instance_variable",
+  "integer",
+  "lambda",
+  "method",
+  "module",
+  "next",
+  "nil",
+  "operator_assignment",
+  "parenthesized_statements",
+  "range",
+  "rational",
+  "redo",
+  "regex",
+  "retry",
+  "return",
+  "scope_resolution",
+  "self",
+  "simple_symbol",
+  "singleton_class",
+  "singleton_method",
+  "string",
+  "string_array",
+  "subshell",
+  "super",
+  "symbol_array",
+  "true",
+  "unary",
+  "unless",
+  "until",
+  "while",
+  "yield",
+];
+
+// TODO: Add more allowable parent types
+const EXPRESSION_STATEMENT_PARENT_TYPES = [
+  "begin",
+  "block",
+  "do",
+  "do_block",
+  "else",
+  "ensure",
+  "heredoc_beginning",
+  "lambda",
+  "method",
+  "program",
+  "singleton_class",
+  "singleton_method",
+  "then",
+];
+
+const mapTypes = ["hash"];
+const listTypes = ["array", "string_array", "symbol_array"];
+
+const assignmentOperators = ["=", '+=', '-=', '*=', '**=', '/=', '||=', '|=', '&&=', '&=', '%=', '>>=', '<<=', '^='];
+const mapKeyValueSeparators = [":", "=>"];
+
+const nodeMatchers: Partial<Record<ScopeType, NodeMatcherAlternative>> = {
+  map: mapTypes,
+  list: listTypes,
+  statement: cascadingMatcher(
+    patternMatcher(...STATEMENT_TYPES),
+    ancestorChainNodeMatcher(
+      [
+        patternFinder(...EXPRESSION_STATEMENT_PARENT_TYPES),
+        patternFinder(...EXPRESSION_TYPES),
+      ],
+      1
+    )
+  ),
+  string: "string",
+  ifStatement: "if",
+  functionCall: "call",
+  comment: "comment",
+  namedFunction: "method",
+  functionName: "method[name]",
+  anonymousFunction: ["lambda", "block", "do_block"],
+  regularExpression: "regex",
+  condition: conditionMatcher("*[condition]"),
+  argumentOrParameter: argumentMatcher(
+    "lambda_parameters",
+    "method_parameters",
+    "block_parameters",
+    "argument_list"
+  ),
+  class: "class",
+  className: "class[name]",
+  collectionKey: trailingMatcher(["pair[key]"], [":"]),
+  name: ["assignment[left]", "operator_assignment[left]", "class[name]", "method[name]"],
+  value: leadingMatcher(
+    ["pair[value]", "assignment[right]", "operator_assignment[right]"],
+    assignmentOperators.concat(mapKeyValueSeparators)
+  ),
+  collectionItem: argumentMatcher(...mapTypes, ...listTypes),
+};
+export const patternMatchers = createPatternMatchers(nodeMatchers);
+
+export function stringTextFragmentExtractor(
+  node: SyntaxNode,
+  _selection: SelectionWithEditor
+) {
+  if (node.type === "string_content") {
```

Comment:
I believe you should add `heredoc_content` to the list of acceptable node types here, and then add a test that you can say "clear round" within a heredoc body